pkgs/by-name/ge/genemichaels/package.nix

```
@@ -1,23 +1,29 @@
-{ lib, fetchFromGitHub, rustPlatform }:
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
 
 rustPlatform.buildRustPackage rec {
   pname = "genemichaels";
-  version = "0.1.21";
+  version = "0.5.7";
+
   src = fetchFromGitHub {
     owner = "andrewbaxter";
-    repo = pname;
-    rev = "158bb8eb705b073d84562554c1a6a63eedd44c6b";
-    hash = "sha256-rAJYukxptasexZzwWgtGlUbHhyyI6OJvSzVxGLBO9vM=";
-  };
-  cargoLock = {
-    lockFile = ./Cargo.lock;
-    outputHashes."markdown-1.0.0-alpha.5" = "sha256-pjIKzXvRKoMfFVIyIXdm+29vvUzCHiJ0rrZgr4K+Ih8=";
+    repo = "genemichaels";
+    rev = "genemichaels-v${version}";
+    hash = "sha256-qPyIC9AbRQI+inBft7Dd5R5v+tXtJcZdiV4lBIBwpyM=";
   };
+
+  cargoHash = "sha256-yPxgYf8N3Dqns/uKhdM++jpzG7zTPhCVVq+7WA9G/SM=";
+
+  cargoBuildFlags = [ "--package ${pname}" ];
```

```suggestion
  cargoBuildFlags = [ "--package genemichaels" ];
  cargoTestFlags = cargoBuildFlags;
```
This makes sure we aren't building other crates when running the checkPhase

---

Hey getchoo

I have a concern with only testing the `genemichaels` package. After version 0.5.7, the author moved the [aargvark](https://github.com/andrewbaxter/aargvark) crate which was being developed separately into the `genemichaels` repository. You can see that happening on this commit: https://github.com/andrewbaxter/genemichaels/commit/957658df78fd1920a385bda9a2919cbeec6f10c5. Their reasoning being that, "They're colocated because of interwtined dependencies and having a single repo makes releasing easier.". After they merged the two repositories, their commits have failed to pass CI because of a failing test in `aargvark`. I was originally going to include the `cargoTestFlags` attribute but if you only target the `genemichaels` package, it will succeed. Because of it's dependence on `aargvark`, I left the `cargoTestFlags` attribute out so that all crates would be built and tested. What do you think the best action is?

---

This is also the reason I only bumped the package to version 0.5.7 instead of 0.5.10 which was the latest at the time. Looks like there is a 0.5.11 now but it still is failing in CI.

---

> I was originally going to include the cargoTestFlags attribute but if you only target the genemichaels package, it will succeed. Because of it's dependence on aargvark, I left the cargoTestFlags attribute out so that all crates would be built and tested. What do you think the best action is?

This was my mistake, you totally made the right decision here. Without context, I thought this was just a common oversight :p

---

Not a problem and totally understandable. Thanks for reviewing and making sure we are keeping with best practices.